client-core/src/main/java/com/influxdb/query/InfluxQLQueryResult.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * @return the results
+     */
+    @Nonnull
+    public List<Result> getResults() {
+        return this.results;
+    }
+
+    /**
+     * Represents one result of an influxQL query.
+     */
+    public final static class Result {
+        private final List<Series> series;
+
+        private final int index;
+
+        public Result(final int index, @Nonnull final List<Series> series) {
+            Arguments.checkNotNull(series, "series");
+
+            this.index = index;
+            this.series = series;
+        }
+
+        /**
+         * @return the index of the result
+         */
+        public int getIndex() {
+            return index;
+        }
+
+        /**
+         * @return the series
+         */
+        @Nonnull
+        public List<Series> getSeries() {
+            return this.series;
+        }
+
+    }
+
+    /**
+     * Represents one series within the {@link Result} of an influxQL query.
+     */
+    public final static class Series {
+        @Nonnull
+        private final Map<String, Integer> columns;
+
+        @Nonnull
+        private final String name;
+
+        private final List<Record> values;
+
+        public Series(final @Nonnull String name, final @Nonnull Map<String, Integer> columns) {
+            Arguments.checkNotNull(name, "name");
+            Arguments.checkNotNull(columns, "columns");
+
+            this.name = name;
+            this.columns = columns;
+            this.values = new ArrayList<>();
+        }
+
+        /**
+         * @return the name
+         */
+        @Nonnull
+        public String getName() {
+            return this.name;
+        }
+
+        /**
+         * @return the columns
+         */
+        @Nonnull
+        public Map<String, Integer> getColumns() {
+            return this.columns;
+        }
+
+
+        /**
+         * @return the values
+         */
+        @Nonnull
+        public List<Record> getValues() {
+            return this.values;
+        }
+
+        public void addRecord(@Nonnull final Record record) {
+            Arguments.checkNotNull(record, "record");
+
+            this.values.add(record);
+        }
+
+        /**
+         * A value extractor is used to convert the string value returned by query into a custom type.
+         */
+        @FunctionalInterface
+        public interface ValueExtractor {
+
+            /**
+             * The value extractor is called for each resulting column to convert the string value returned by query
+             * into a custom type.
+             *
+             * @param columnName the name of the column
+             * @param rawValue the string value returned from the query
+             * @param resultIndex the index of the result
+             * @param seriesName the name of the series
+             * @return the converted string value
+             */
+            Object extractValue(String columnName, String rawValue, int resultIndex, String seriesName);
```

Comment:
Please add `@Nullable` and `@Nonnull` annotations to method parameters and result.